apps/opik-backend/src/main/java/com/comet/opik/domain/OpenTelemetryMapper.java

```diff
@@ -47,39 +51,121 @@ public static com.comet.opik.api.Span toOpikSpan(Span otelSpan, UUID opikTraceId
+     *
+     * @param spanBuilder the span builder where we will be injecting the extracted values
+     * @param attributes the list of span attributes extracted from the otel payload
+     * @param integrationName the name of the integration sending the spans (can be empty)
+     */
+    public static void enrichSpanWithAttributes(SpanBuilder spanBuilder, List<KeyValue> attributes,
+            String integrationName) {
+        Map<String, Integer> usage = new HashMap<>();
+        ObjectNode input = JsonUtils.MAPPER.createObjectNode();
+        ObjectNode output = JsonUtils.MAPPER.createObjectNode();
+        ObjectNode metadata = JsonUtils.MAPPER.createObjectNode();
+
+        if (StringUtils.isNotEmpty(integrationName)) {
+            metadata.put("integration", integrationName);
+        }
 
         // Iterate over each attribute key-value pair
         attributes.forEach(attribute -> {
             var key = attribute.getKey();
             var value = attribute.getValue();
 
-            switch (value.getValueCase()) {
-                case STRING_VALUE -> node.put(key, StringEscapeUtils.unescapeJson(value.getStringValue()));
-                case INT_VALUE -> node.put(key, value.getIntValue());
-                case DOUBLE_VALUE -> node.put(key, value.getDoubleValue());
-                case BOOL_VALUE -> node.put(key, value.getBoolValue());
-                default -> log.warn("Unsupported attribute: {}", attribute);
-            }
+            OpenTelemetryMappingRule.findRule(key).ifPresentOrElse(rule -> {
+                Optional.ofNullable(rule.getSpanType()).ifPresent(spanBuilder::type);
+
+                switch (rule.getOutcome()) {
+                    case MODEL :
+                        spanBuilder.model(value.getStringValue());
+                        break;
+
+                    case USAGE :
+                        // usage might appear as int values or an json object
+                        if (value.hasIntValue()) {
+                            var actualKey = key.substring(rule.getRule().length());
+                            usage.put(actualKey, (int) value.getIntValue());
+                        } else {
+                            JsonNode usageNode = JsonUtils.getJsonNodeFromString(value.getStringValue());
+                            if (usageNode.isTextual()) {
+                                usageNode = JsonUtils.getJsonNodeFromString(usageNode.asText());
+                            }
+                            usageNode.fields().forEachRemaining(entry -> {
+                                if (entry.getValue().isNumber()) {
+                                    usage.put(entry.getKey(), entry.getValue().intValue());
+                                } else {
+                                    log.warn("Unrecognized usage attribute {}: {}", entry.getKey(), entry.getValue());
+                                }
+                            });
```

Comment:
Can we extract this to private methods? Just to make the switch case simpler to read